tests/client/core/components/TestHoverIntent.js

```diff
@@ -0,0 +1,115 @@
+import React from 'react';
+import {
+  findRenderedComponentWithType,
+  renderIntoDocument,
+  Simulate,
+} from 'react-addons-test-utils';
+import {
+  findDOMNode,
+} from 'react-dom';
+
+import HoverIntent from 'core/components/HoverIntent';
+
+function renderHoverIntent({ ...props }) {
+  const sensitivity = 5;
+  const interval = 50;
+  const hoverIntent = findRenderedComponentWithType(renderIntoDocument(
+    <HoverIntent {...props} sensitivity={sensitivity} interval={interval}>
+      <span id="text-content">Text content</span>
+    </HoverIntent>
+  ), HoverIntent);
+  const innerElement = findDOMNode(hoverIntent);
+
+  return { hoverIntent, innerElement };
+}
+
+describe('<HoverIntent />', () => {
+  let hoverIntent;
+  let innerElement;
+  let props;
+  let clock;
+
+  beforeEach(() => {
+    clock = sinon.useFakeTimers();
+    props = {
+      onHoverIntent: sinon.spy(),
+      onHoverIntentEnd: sinon.spy(),
+    };
+    const rendered = renderHoverIntent(props);
+    hoverIntent = rendered.hoverIntent;
+    innerElement = rendered.innerElement;
+  });
+
+  afterEach(() => {
+    clock.restore();
+  });
+
+  it('runs onHoverIntent if mouse slows/stops on inner element', () => {
+    Simulate.mouseOver(innerElement, { clientX: 1, clientY: 2 });
+
+    clock.tick(75);
+    Simulate.mouseMove(innerElement, { clientX: 10, clientY: 0 });
+
+    clock.tick(50);
+    Simulate.mouseMove(innerElement, { clientX: 10, clientY: 1 });
+
+    clock.tick(50);
+    Simulate.mouseOut(innerElement);
+
+    const { onHoverIntent, onHoverIntentEnd } = props;
+    assert.ok(onHoverIntent.calledOnce);
```

Comment:
Can you assert after each tick? I think this test would pass if it was just mapping `onHoverIntent` to `onHover`.

```jsx
const { onHoverIntent, onHoverIntentEnd } = props;
Simulate.mouseOver(innerElement, { clientX: 1, clientY: 2 });
clock.tick(75);
assert.isNotOk(onHoverIntent.called);

Simulate.mouseMove(innerElement, { clientX: 10, clientY: 0 });
clock.tick(50);
assert.ok(onHoverIntent.calledOnce);

Simulate.mouseMove(innerElement, { clientX: 10, clientY: 1 });
clock.tick(50);
assert.ok(onHoverIntent.calledOnce);
assert.isNotOk(onHoverIntentEnd.called);

Simulate.mouseOut(innerElement);
assert.ok(onHoverIntent.calledOnce);
assert.ok(onHoverIntentEnd.calledOnce);